mojave/code/modules/mob/living/carbon/human/human_armor.dm

```diff
@@ -0,0 +1,91 @@
+/mob/living/carbon/human/getsubarmor(def_zone, d_type)
+	if(!def_zone)
+		//no averaging values when no bodypart is specified, that's stupid
+		return 0
+	if(isbodypart(def_zone))
+		return checksubarmor(def_zone, d_type)
+	var/obj/item/bodypart/affecting = get_bodypart(check_zone(def_zone))
+	if(affecting)
+		//If a specific bodypart is targeted, check how that bodypart is protected and return the value.
+		return checksubarmor(affecting, d_type)
+
+//We only get the most superficial edge protection, no stacking!
+/mob/living/carbon/human/get_edge_protection(def_zone)
+	var/obj/item/bodypart/affecting
+	if(def_zone)
+		if(isbodypart(def_zone))
+			affecting = def_zone
+		else
+			affecting = get_bodypart(check_zone(def_zone))
+
+	//No support for averaging out edge protection because it'd be fucking stupid
+	if(!affecting)
+		return 0
+
+	var/list/clothings = clothingonpart(affecting)
+	for(var/obj/item/clothing in clothings)
+		return clothing.subarmor.getRating(EDGE_PROTECTION)
+	return physiology.subarmor.getRating(EDGE_PROTECTION)
+
+//We only get the most superficial armor flags, no stacking!
+/mob/living/carbon/human/get_subarmor_flags(def_zone)
+	var/obj/item/bodypart/affecting
+	if(def_zone)
+		if(isbodypart(def_zone))
+			affecting = def_zone
+		else
+			affecting = get_bodypart(check_zone(def_zone))
+
+	//if you don't specify a bodypart then only your physiology matters
+	if(!affecting)
+		return physiology.subarmor.subarmor_flags
+
+	var/list/clothings = clothingonpart(affecting)
+	for(var/obj/item/clothing as anything in clothings)
```

Comment:
Same deal as edge protection above.

Reply:
subarmor_flags can stay like this, it should work fine from the order i saw in clothingonpart